### PR TITLE
chore: refactoring

### PR DESCRIPTION
**Pull request by lmoe**

### CODE REVIEW

File: packages/isc/sandbox_interface.go [hunk 1]
```diff
@@ -402,6 +402,10 @@ func (s StateAnchor) Hash() hashing.HashValue {
 }
 
 func (s StateAnchor) Equals(input *StateAnchor) bool {
+	if input == nil {
+		return false
+	}
+
```
Comment (dessaya) [inline packages/isc/sandbox_interface.go:408]: This looks fishy. Is it considered ok for the pointer to be nil?

File: tools/wasp-cli/chain/deploy.go [hunk 2]
```diff
@@ -5,6 +5,9 @@ package chain
 
 import (
 	"context"
+	"github.com/iotaledger/hive.go/kvstore/mapdb"
+	"github.com/iotaledger/wasp/packages/state/indexedstore"
+	"github.com/iotaledger/wasp/packages/transaction"
 	"os"
```
Comment (dessaya) [inline tools/wasp-cli/chain/deploy.go]: Pls see if you can tell your IDE to place the imports in the right order :rofl:

**Review by dessaya — APPROVED** [review]

File: tools/wasp-cli/chain/deploy.go (hunk 2 above)
Comment (lmoe) [reply tools/wasp-cli/chain/deploy.go]: Jesus christ, not again

File: packages/isc/sandbox_interface.go (hunk 1 above)
Comment (lmoe) [reply packages/isc/sandbox_interface.go:408]: It's getting used quite often and the value is directly passed into it. Sometimes the pointer can be nil it seems.

Comment (lmoe) [reply packages/isc/sandbox_interface.go:408]: Mostly in the consensus / state parts. Where we have callbacks receiving a pointer. In some cases it was nil there, yeah

File: tools/wasp-cli/chain/deploy.go (hunk 2 above)
Comment (lmoe) [reply tools/wasp-cli/chain/deploy.go]: fixed :D

File: clients/iscmove/iscmoveclient/client_anchor.go [hunk 3]
```diff
@@ -37,7 +37,8 @@ func (c *Client) FindCoinsForGasPayment(
 
 func (c *Client) StartNewChain(
 	ctx context.Context,
-	signer cryptolib.Signer,
+	cryptolibSigner cryptolib.Signer,
+	chainOwnerAddress *cryptolib.Address,
```
Comment (howjmay) [inline clients/iscmove/iscmoveclient/client_anchor.go:41]: So the signer may not be the be the chain owner?

Comment (lmoe) [reply clients/iscmove/iscmoveclient/client_anchor.go:41]: Yes, if you call this function with the wasp-cli you call it with your own key, not the Committees key. Which requires you to transfer the ownership of the Anchor afterwords. 

https://github.com/iotaledger/isc-private/blob/develop/tools/wasp-cli/chain/deploy.go#L121
https://github.com/iotaledger/isc-private/blob/develop/tools/wasp-cli/chain/deploy.go#L134

**Pull request merged**